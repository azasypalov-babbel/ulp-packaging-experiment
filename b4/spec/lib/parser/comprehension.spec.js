import example from '../../fixtures/unparsed/comprehension.json';
import parse from '../../../src/lib/parser/comprehension';

describe('Parser', () => {
  describe('Comprehension', () => {
    const parsed = parse(example);

    test('maps the data onto a different format', () => {
      expect(parsed).toMatchSnapshot();
    });

    test('is formatting data in a specific order', () => {
      const expectedParsedFormatType = ['title', 'description', 'Phrase', 'Phrase', 'Phrase', 'Task', 'Task', 'Task'];
      const parsedTypes = parsed.traineritems.map((item) => item.type);

      expect(parsedTypes).toEqual(expectedParsedFormatType);
    });

    describe('first parsed item', () => {
      test('is "title"', () => {
        expect(parsed.traineritems[0].type).toBe('title');
      });

      test('uses the title of unparsed input as "l1_text"', () => {
        expect(parsed.traineritems[0].l1_text).toBe(example.title);
      });
    });


    describe('second parsed item', () => {
      test('is "description"', () => {
        expect(parsed.traineritems[1].type).toBe('description');
      });

      test('uses the description of unparsed input as "l1_text"', () => {
        expect(parsed.traineritems[1].l1_text).toBe(example.description);
      });
    });

    describe('when dictate flag is true', () => {
      const audioComprehension = Object.assign({}, example, {
        dictate: true
      });

      test('maps "mode" to "AudioDialog"', () => {
        expect(parse(audioComprehension).mode).toBe('AudioDialog');
      });
    });

    describe('when dictate flag is false', () => {
      const textComprehension = Object.assign({}, example, {
        dictate: false
      });

      test('maps "mode" to "Text"', () => {
        expect(parse(textComprehension).mode).toBe('Text');
      });
    });

    describe('when "translation_visibility" is "full"', () => {
      const fullTranslationComprehension = Object.assign({}, example, {
        translation_visibility: 'full'
      });

      test('maps "translation" to "Show"', () => {
        expect(parse(fullTranslationComprehension).translation).toBe('Show');
      });
    });

    describe('when "translation_visibility" is "partial"', () => {
      const partialTranslationComprehension = Object.assign({}, example, {
        translation_visibility: 'partial'
      });

      test('maps "translation" to "Icon"', () => {
        expect(parse(partialTranslationComprehension).translation).toBe('Icon');
      });
    });

    describe('when "translation_visibility" is "none"', () => {
      const noneTranslationComprehension = Object.assign({}, example, {
        translation_visibility: 'none'
      });

      test('maps "translation" to "Hide"', () => {
        expect(parse(noneTranslationComprehension).translation).toBe('Hide');
      });
    });
  });
});
