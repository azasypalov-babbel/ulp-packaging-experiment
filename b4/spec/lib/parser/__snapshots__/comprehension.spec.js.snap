// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parser Comprehension maps the data onto a different format 1`] = `
Object {
  "component": "comprehension",
  "mode": "Text",
  "traineritems": Array [
    Object {
      "id": 0,
      "image_id": null,
      "info_text": "",
      "l1_text": "This is some title",
      "l2_text": "",
      "sound_id": null,
      "speaker_role": null,
      "type": "title",
    },
    Object {
      "id": 0,
      "image_id": null,
      "info_text": "",
      "l1_text": "Audrey is sitting at the computer reading the latest news about the festival...",
      "l2_text": "",
      "sound_id": null,
      "speaker_role": null,
      "type": "description",
    },
    Object {
      "id": "e3a642e6e18cdb668b6639a3946ffcc4",
      "image_id": null,
      "info_text": "",
      "l1_text": "**Cannes : small piece of gossip before the awards ceremony**",
      "l2_text": "**Cannes : petit cancan avant la remise des prix**",
      "sound_id": null,
      "speaker_role": "NONE",
      "type": "Phrase",
    },
    Object {
      "id": "1a494c8a983a86e0258f4916cd1fa074",
      "image_id": null,
      "info_text": "",
      "l1_text": "from the \\"Gazette de Cannes\\"",
      "l2_text": "par \\"La Gazette de Cannes\\"",
      "sound_id": null,
      "speaker_role": "NONE",
      "type": "Phrase",
    },
    Object {
      "id": "9a1115b582b265b06692f038ecb3b2fa",
      "image_id": null,
      "info_text": "",
      "l1_text": "The festival ends on Saturday with the showing of \\"Mina\\", by Jean-Luc Godasse.",
      "l2_text": "Le festival s'ach√®ve samedi avec la projection de \\"Mina\\", de Jean-Luc Godasse.",
      "sound_id": null,
      "speaker_role": "NONE",
      "type": "Phrase",
    },
    Object {
      "id": "63e63811c7a714039e880daeb31e269b",
      "image_id": null,
      "info_text": "",
      "l1_text": "When does the festival finish? ((Sunday|This evening|*Saturday))",
      "l2_text": "",
      "sound_id": null,
      "speaker_role": null,
      "type": "Task",
    },
    Object {
      "id": "a47f8ccdabe18401d9ffc2f8525e3f81",
      "image_id": null,
      "info_text": "",
      "l1_text": "Who is the leading actor of the film \\"Mina\\"? ((a star|*a dog|Jean-Luc Godasse))",
      "l2_text": "",
      "sound_id": null,
      "speaker_role": null,
      "type": "Task",
    },
    Object {
      "id": "4f072cf656bf3a36d888016bda1f515c",
      "image_id": null,
      "info_text": "",
      "l1_text": "Will the dog make an appearance on the red carpet? ((*maybe|definitely not|yes, wearing a dress for certain))",
      "l2_text": "",
      "sound_id": null,
      "speaker_role": null,
      "type": "Task",
    },
  ],
  "translation": "Icon",
}
`;
